File: Problem20.py
#Problem: Find the sum of the digits in the number 100!

def fat(n):
	if(n == 1):
		return 1
	else:
		return n * fat(n-1)

		
sum = 0		
fat = fat(100)
fat = str(fat)
for i in range(len(fat)):
	sum += int(fat[i])

	
print("The sum is",sum) #outputs 648

File: Problem20.java

import java.math.BigDecimal;

public class Problem20 {

		
	//Solution in Java for Project Euler number 20 problem
	public BigDecimal factorial(int n){
		BigDecimal fat = new BigDecimal("1");
		for(int i = 2; i <= n;i++){
			fat = fat.multiply(new BigDecimal(i));
		}
		
		return fat;
	}
	
	
	public int sumDigits(String strNumber){
		int sum = 0;
		for(int i = 0; i < strNumber.length();i++){
			sum+= Integer.parseInt(Character.toString(strNumber.charAt(i)));
		}
		return sum;
	}
	
	public static void main(String[] args) {

		int sum = new Problem20().sumDigits(new Problem20().factorial(100).toString());
		System.out.println("The sum is " + sum);
	}

}
